engine/src/xml/mod.rs

```diff
@@ -61,14 +61,24 @@ impl<'de> Visitor<'de> for VersionVisitor {
     where
         E: de::Error,
     {
-        if let Ok((major, minor, patch)) = scan_fmt!(&v.to_string(), "{d}.{d}.{d}", u32, u32, u32) {
+        if let Ok((major, minor, patch)) = scan_fmt!(&v, "{d}.{d}.{d}", u32, u32, u32) {
```

Comment:
This entire code must be purged I think 😂 I wrote it before knowing Rust, so there is a lot to be desired from it. Thank you for improving it!

Reply:
I'm actually enjoying this 😀